pkg/bgpv1/manager/reconcilerv2/paths.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"context"
+	"maps"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/instance"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+)
+
+// PathMap is a map of paths indexed by the NLRI string
+type PathMap map[string]*types.Path
+
+// AFPathsMap is a map of paths per address family, indexed by the family
+type AFPathsMap map[types.Family]PathMap
+
+type ReconcileAFPathsParams struct {
+	Logger       logrus.FieldLogger
+	Ctx          context.Context
+	Instance     *instance.BGPInstance
+	DesiredPaths AFPathsMap
+	CurrentPaths AFPathsMap
+}
+
+type reconcilePathsParams struct {
+	Logger                logrus.FieldLogger
+	Ctx                   context.Context
+	Instance              *instance.BGPInstance
+	CurrentAdvertisements PathMap
+	ToAdvertise           PathMap
+}
+
+// ReconcileAFPaths reconciles BGP advertisements per address family. It will consume desired and current paths (AFPathsMap)
+// and will return the outcome of the reconciliation.
+func ReconcileAFPaths(rp *ReconcileAFPathsParams) (AFPathsMap, error) {
+	runningAFPaths := make(AFPathsMap)
```

Comment:
Current and desired are the states passed by callers. I kept `running` as the state which will get mutated. Idea was not to mutate maps which are passed by the caller, instead create a copy of current map and mutate it as we add/remove items based on diff. Result of which is called `running`. Caller can decide what it want's to do with the returned map, especially if there is an error returned as well. 

Naming is debatable - instead of `running` what would you prefer ? I didn't want to name it `current` as it can be confusing that we are mutating caller's map directly, since it's also called current.

Reply:
@harsimran-pabla thanks for the add'l context. I understand your desire to distinguish between the passed-in map for the `current` state. On the other hand, a reader needs to distinguish between `current` and `running`, two nearly identical terms. I'm fine proceeding as-is.